devices/usbc-cdc/ctrl.c

```diff
@@ -52,10 +55,13 @@ enum {
 
 static int ctrl_allocBuff(int endpt, int dir)
 {
-	ctrl_common.data->endpts[endpt].buf[dir].buffer = usbclient_allocBuff(USB_BUFFER_SIZE);
-
-	if (ctrl_common.data->endpts[endpt].buf[dir].buffer == NULL)
-		return -ENOMEM;
+	/* Allocate buffer for the first time (initially is NULL) */
+	if (ctrl_common.data->endpts[endpt].buf[dir].buffer == NULL) {
```

Comment:
Is it possible for `ctrl_common.data->endpts[endpt].buf[dir].buffer` to not be `NULL` here?

Reply:
Yes of course, allocation is done once at init time, buffer pointers does not change in time, further calls to `allocBuff(..)` for particular endpoint returns already allocated memory room for selected `endpt`, this is done e.g. for each reset received from USB, e.g. initiated by host, before enumeration and in other cases. Previous implementation caused bus fault, when USB reset was received. Maybe the the functions name are misleading and should be changed to sth like `*_resetAlloc*(..)` than?